lib/ansible/galaxy/collection.py

```diff
@@ -399,6 +399,13 @@ def from_path(b_path, force, parent=None):
                 except ValueError:
                     raise AnsibleError("Collection file at '%s' does not contain a valid json string."
                                        % to_native(b_file_path))
+        if not info and fallback_metadata:
+            b_galaxy_path = os.path.join(b_path, b'galaxy.yml')
```

Comment:
I'm not sure if we want to support `galaxy.yaml` as well. We do that in certain places so it may be good to add that.

Reply:
For ansible-doc this was added:
`lib/ansible/collections/__init__.py:B_FLAG_FILES = frozenset([b'MANIFEST.json', b'galaxy.yml'])`

I think we could just centralize usage here in case of future expansion (or even move to constants.py)

Reply:
@samdoran I don't see `galaxy.yaml` supported anywhere for `ansible-galaxy collection` yet. Maybe it should be implemented in a separate PR though, so that it can be added for `build` at the same time rather than piecemeal.

@bcoca +1 to centralizing

Reply:
@s-hertel I may have just imagined it, or maybe I saw this in `execute_install()` https://github.com/ansible/ansible/blob/25c5388fdec9e56517a93feb5e8d485680946c25/lib/ansible/cli/galaxy.py#L1021-L1021

Anyway, this should be a separate PR and centralized.